app/projects/forms.py

```diff
@@ -743,6 +746,51 @@ def __init__(self, *args, **kwargs):
 
         """ ----------------------------------------------------- """
 
+        """ Display cost values in form according to given exchange rate, costs will be saved as USD in the database.
+        Set default cost values depending on the asset type"""
+        # TODO add main grid to this when implemented (needs some special care because energy price is saved as string)
+        cp_nigeria_defaults = {
+            "pv_plant": {"lifetime": 25, "capex_var": 477, "opex_fix": 10},
+            "diesel_generator": {
+                "lifetime": 8,
+                "capex_var": 400,
+                "opex_fix": 25,
+                "opex_var": 0.03,
+                "opex_var_extra": 1.11,
+                # TODO connect to https://www.globalpetrolprices.com/Nigeria/diesel_prices/ and use this as value once per day
+                # ie solution with entry in the DB because the date needs to be linked to the price to be updated if the date is different
+                "efficiency": 0.25,
+                "soc_min": 0.0,
+                "soc_max": 1.0,
+            },
+            "capacity": {
+                "lifetime": 8,
+                "capex_var": 331,
+                "opex_fix": 10,
+                "crate": 1,
+                "soc_min": 0.8,
+                "soc_max": 1,
+                "efficiency": 0.9,
+            },
+        }
+
+        if self.asset_type_name in cp_nigeria_defaults:
+            if self.existing_asset is None:
+                default_values = cp_nigeria_defaults[self.asset_type_name]
+                for field, initial_value in default_values.items():
+                    if field in self.currency_fields:
+                        self.initial[field] = round((initial_value * self.exchange_rate), 2)
+                    else:
+                        self.initial[field] = initial_value
+            else:
+                for field in self.currency_fields:
+                    if hasattr(self.existing_asset, field):
+                        self.initial[field] = getattr(self.existing_asset, field) * self.exchange_rate
+                        if field == "opex_var_extra":
+                            self.initial[field] = self.initial["opex_var_extra"] * ENERGY_DENSITY_DIESEL
+
+                        self.initial[field] = round(self.initial[field], 2)
+

```

Comment:
I think I overlooked that , any specific reason for doing this that way instead of Making a CPAsset class which inherit the AssetCreateForm and make the CP asset inherit from this class instead of AssetCreateForm? Then one overload the __init__ call, running first the __init__ of the parent class and then the code specific to the child class